src/Sylius/Component/Core/Inventory/Operator/OrderInventoryOperator.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Inventory\Operator;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class OrderInventoryOperator implements OrderInventoryOperatorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function cancel(OrderInterface $order)
+    {
+        if (OrderPaymentStates::STATE_PAID === $order->getPaymentState()) {
+            $this->giveBack($order);
+
+            return;
+        }
+        $this->release($order);
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function hold(OrderInterface $order)
+    {
+        /** @var OrderItemInterface $orderItem */
+        foreach ($order->getItems() as $orderItem) {
+            $variant = $orderItem->getVariant();
+
+            if ($variant->isTracked()) {
+                $variant->setOnHold($variant->getOnHold() + $orderItem->getQuantity());
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sell(OrderInterface $order)
+    {
+        /** @var OrderItemInterface $orderItem */
+        foreach ($order->getItems() as $orderItem) {
+            $variant = $orderItem->getVariant();
+
+            if ($variant->isTracked()) {
+
+                Assert::greaterThanEq(
```

Comment:
The more specific exception than `\InvalidArgumentException` would be nicer to work with.